uaclient/cli.py

```diff
@@ -103,26 +101,66 @@ def __init__(
         )
 
         self.base_desc = base_desc
-        self.non_beta_services_desc = non_beta_services_desc
-        self.beta_services_desc = beta_services_desc
 
     def error(self, message):
         self.print_usage(sys.stderr)
         self.exit(2, message + "\n")
 
     def print_help(self, file=None, show_all=False):
-        desc_vars = [
-            self.base_desc,
-            self.non_beta_services_desc,
-            self.beta_services_desc,
-        ]
-        if any(desc_vars):
-            services = sorted(self.non_beta_services_desc)
+        if self.base_desc:
+            non_beta_services_desc, beta_services_desc = (
+                self._get_service_descriptions()
+            )
+            service_descriptions = sorted(non_beta_services_desc)
             if show_all:
-                services = sorted(services + self.beta_services_desc)
-            self.description = "\n".join([self.base_desc] + services)
+                service_descriptions = sorted(
+                    service_descriptions + beta_services_desc
+                )
+            self.description = "\n".join(
+                [self.base_desc] + service_descriptions
+            )
         super().print_help(file=file)
 
+    @staticmethod
+    def _get_service_descriptions() -> Tuple[List[str], List[str]]:
+        service_line_tmpl = " - {name}: {description}{url}"
+        non_beta_services_desc = []
+        beta_services_desc = []
+
+        resources = contract.get_available_resources(config.UAConfig())
+        for resource in resources:
+            ent_cls = entitlements.entitlement_factory(resource["name"])
+            if ent_cls:
+                # Because we don't know the presentation name if unattached
+                presentation_name = resource.get(
+                    "presentedAs", resource["name"]
+                )
+                if ent_cls.help_doc_url:
+                    url = " ({})".format(ent_cls.help_doc_url)
+                else:
+                    url = ""
+                service_line = service_line_tmpl.format(
+                    name=presentation_name,
+                    description=ent_cls.description,
+                    url=url,
+                )
+                if len(service_line) <= 80:
+                    service_info = [service_line]
+                else:
+                    wrapped_words = []  # type: List[str]
+                    line = service_line
+                    while len(line) > 80:
+                        [line, wrapped_word] = line.rsplit(" ", 1)
+                        wrapped_words.insert(0, wrapped_word)
+                    service_info = [line + "\n   " + " ".join(wrapped_words)]
```

Comment:
Since we're touching this logic, we can probably replace this chunk with something like
```python
textwrap.wrap(
  service_line,
  width=defaults.PRINT_WRAP_WIDTH,
  subsequent_indent="   ",
  break_long_words=False,
  break_on_hyphens=False
)
```

Reply:
indeed